panic when ssh public key or ssh key name not exists on DO Provider

**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

Version 1.14.1

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

v1.14.7

**3. What cloud provider are you using?**

DigitalOcean

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

kops update cluster kops.example.com --yes

**5. What happened after the commands executed?**

panic: runtime error: index out of range

**6. What did you expect to happen?**

should return an error when `sshPublicKeys` or `Cluster.Spec.SSHKeyName` not exists

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

**9. Anything else do we need to know?**

We must validate `sshPublicKeys` or `Cluster.Spec.SSHKeyName` and return an error when not exists

/assign

--- a/upup/pkg/fi/cloudup/apply_cluster.go
+++ b/upup/pkg/fi/cloudup/apply_cluster.go
@@ -385,6 +385,10 @@ func (c *ApplyClusterCmd) Run() error {
 				return fmt.Errorf("DigitalOcean support is currently (very) alpha and is feature-gated. export KOPS_FEATURE_FLAGS=AlphaAllowDO to enable it")
 			}
 
+			if len(sshPublicKeys) == 0 && c.Cluster.Spec.SSHKeyName == "" {
+				return fmt.Errorf("SSH public key must be specified when running with DigitalOcean (create with `kops create secret --name %s sshpublickey admin -i ~/.ssh/id_rsa.pub`)", cluster.ObjectMeta.Name)
+			}
+
 			modelContext.SSHPublicKeys = sshPublicKeys
 
 			l.AddTypes(map[string]interface{}{

